File: lang/ruby/src/fun/vac/algorithms/search/BinarySearch.rb
module BinarySearch

    def self.find(arr, key)
        lo = 0
        hi = arr.length

        mid = 0

        while lo < hi
            mid = lo + ((hi - lo) >> 1)
            case
            when key < arr[mid]
                hi = mid
            when key > arr[mid]
                lo = mid + 1
            else
                return mid
            end
        end

        return arr.length
    end
end

File: lang/php/src/fun/vac/algorithms/search/BinarySearch.php
<?php
    namespace fun\vac\algorithms\search\BinarySearch;

    function find(array &$arr, $key): int {
        $lo = 0;
        $hi = count($arr);

        $mid = 0;

        while ($lo < $hi) {
            $mid = $lo + (($hi - $lo) >> 1);
            if ($key < $arr[$mid]) {
                $hi = $mid;
            } else if ($key > $arr[$mid]) {
                $lo = $mid + 1;
            } else {
                return $mid;
            }
        }

        return count($arr);
    }
?>
